src/engine/evaluator/index.js

```diff
@@ -41,26 +40,21 @@ function match(matchingResult: boolean, bucketingKey: string, seed: number, trea
 function evaluatorContext(matcherEvaluator: Function, treatments: Treatments, label: string, conditionType: string): Function {
 
   function evaluator(key: SplitKey, seed: number, trafficAllocation: number, trafficAllocationSeed: number, attributes: ?Object, algo: ?number): AsyncValue<?Evaluation> {
-    // the key could be a string or KeyDTO it should return a keyDTO.
-    const {
-      matchingKey,
-      bucketingKey
-    } = keyParser(key);
 
     // Whitelisting has more priority than traffic allocation, so we don't apply this filtering to those conditions.
-    if (conditionType === 'ROLLOUT' && !engine.shouldApplyRollout(trafficAllocation, bucketingKey, trafficAllocationSeed, algo)) {
+    if (conditionType === 'ROLLOUT' && !engine.shouldApplyRollout(trafficAllocation, key.bucketingKey, trafficAllocationSeed, algo)) {
       return;
     }
 
     // matcherEvaluator could be Async, this relays on matchers return value, so we need
     // to verify for thenable before play with the result
-    const matches = matcherEvaluator(matchingKey, attributes);
+    const matches = matcherEvaluator(key.matchingKey, attributes);
 
     if (thenable(matches)) {
-      return matches.then(result => match(result, bucketingKey, seed, treatments, label, algo));
+      return matches.then(result => match(result, key.bucketingKey, seed, treatments, label, algo));
```

Comment:
Lets review this because matching mechanism should use always the `matchingKey`, `bucketingKey` should be used only in the bucket calculation.